File: Python/Servicio.py
from datetime import datetime

class Servicio:
    def __init__(self, usuario, vehiculo, guardia_entrada, guardia_patio, fecha_entrada, fecha_salida):
        self.usuario = usuario
        self.vehiculo = vehiculo
        self.guardia_entrada = guardia_entrada
        self.guardia_patio = guardia_patio
        self.fecha_entrada = fecha_entrada
        self.fecha_salida = fecha_salida

    # Metodos Getters
    def get_usuario(self):
        return self.usuario
    
    def get_vehiculo(self):
        return self.vehiculo
    
    def get_guardia_entrada(self):
        return self.guardia_entrada
    
    def get_guardia_patio(self):
        return self.guardia_patio
    
    def get_fecha_entrada(self):
        return self.fecha_entrada
    
    def get_fecha_salida(self):
        return self.fecha_salida
    

    # Metodos Setters
    def set_usuario(self, usuario):
        self.usuario = usuario

    def set_vehiculo(self, vehiculo):
        self.vehiculo = vehiculo

    def set_guardia_entrada(self, guardia_entrada):
        self.guardia_entrada = guardia_entrada

    def set_guardia_patio(self, guardia_patio):
        self.guardia_patio = guardia_patio
    
    def set_fecha_entrada(self, fecha_entrada):
        self.fecha_entrada = fecha_entrada
    
    def set_fecha_salida(self, fecha_salida):
        self.fecha_salida = fecha_salida
    

    def __str__(self):
        return (f"Servicio(Usuario= '{self.usuario.__str__()}', "
                f"Vehiculo='{self.vehiculo.__str__()}', Guardia Entrada = '{self.guardia_entrada.__str__()}', "
                f"Guardia Patio = '{self.guardia_patio.__str__()}', Fecha Entrada = '{self.fecha_entrada}', "
                f"Fecha Salida = '{self.fecha_salida}')")
        

File: Java/Servicio.java
import java.util.Date;

public class Servicio {
    private int id;
    private static int contadorId;
    private Date fechaEntrada;
    private Date fechaSalida;
    private Usuario conductor;
    private Vehiculo vehiculo;
    private Guardia guardiaEntrada;
    private Guardia guardiaPatio;

    public Servicio(Usuario usuario, Vehiculo vehiculo, Guardia guardiaEntrada, Guardia guardiaPatio, Date fechaEntrada, Date fechaSalida){
        this.id = ++Servicio.contadorId;
        this.conductor = usuario;
        this.vehiculo = vehiculo;
        this.guardiaEntrada = guardiaEntrada;
        this.guardiaPatio = guardiaPatio;
        this.fechaEntrada = fechaEntrada;
        this.fechaSalida = fechaSalida;
    }

    public int getId() {
        return id;
    }

    public static int getContadorId() {
        return contadorId;
    }

    public static void setContadorId(int contadorId) {
        Servicio.contadorId = contadorId;
    }

    public Date getFechaEntrada() {
        return fechaEntrada;
    }

    public void setFechaEntrada(Date fechaEntrada) {
        this.fechaEntrada = fechaEntrada;
    }

    public Date getFechaSalida() {
        return fechaSalida;
    }

    public void setFechaSalida(Date fechaSalida) {
        this.fechaSalida = fechaSalida;
    }

    public Usuario getConductor() {
        return conductor;
    }

    public void setConductor(Usuario conductor) {
        this.conductor = conductor;
    }

    public Vehiculo getVehiculo() {
        return vehiculo;
    }

    public void setVehiculo(Vehiculo vehiculo) {
        this.vehiculo = vehiculo;
    }

    public Guardia getGuardiaEntrada() {
        return guardiaEntrada;
    }

    public void setGuardiaEntrada(Guardia guardiaEntrada) {
        this.guardiaEntrada = guardiaEntrada;
    }

    public Guardia getGuardiaPatio() {
        return guardiaPatio;
    }

    public void setGuardiaPatio(Guardia guardiaPatio) {
        this.guardiaPatio = guardiaPatio;
    }

    @Override
    public String toString() {
        return "Servicio [id=" + this.getId() + ", fechaEntrada=" + this.getFechaEntrada() + ", fechaSalida=" + this.getFechaSalida() + ", conductor="
                + this.getConductor() + ", vehiculo=" + this.getVehiculo() + ", guardiaEntrada=" + this.getGuardiaEntrada() + ", guardiaPatio="
                + this.getGuardiaPatio() + "]";
    }

    
}
